app/models/changeset.rb

```diff
@@ -70,6 +71,11 @@ def find_comparison
     if empty?
       NullComparison.new(nil)
     else
+      # We try to fetch the SHA of branch
```

Comment:
something like this ?
`# for branches that need review we make sure to always get the correct cache, others might get a outdated changeset if they are reviewed with different shas`